duck/ornithology.py

```diff
@@ -9,27 +9,58 @@
 ANY = mock.ANY
 
 
-class Instance(object):
-    """A class that checks equality based on type.
+class Predicate(object):
+    """An object considered to be equal to anything that passes its predicate
+    function.
 
-    Instances of Instance have an .__eq__ method that returns
-    True if and only if the other comparison object is an
-    instance of the provided class (as defined by isinstance).
+    Args:
+        predicate (Callable): The predicate to use to test.
     """
-    def __init__(self, _class):
-        self._class = _class
+    def __init__(self, predicate):
+        self._predicate = predicate
 
     def __eq__(self, other):
-        return isinstance(other, self._class)
+        return self._predicate(other)
 
     def __ne__(self, other):
         return not self == other
 
+
+class Instance(Predicate):
+    """An object considered equal to any instance of the correct type.
+
+    Args:
+        class_ (type): The class to be checked against.
+    """
+    def __init__(self, class_):
+        self._class = class_
+        super(Instance, self).__init__(lambda t: isinstance(t, class_))
+
     def __repr__(self):
         return '<Instance: {0}>'.format(self._class.__name__)
 
 
+class Needle(Predicate):
+    """An object considered equal to any instance in which the compared
+    value is contained within the presented object.
+
+    Args:
+        haystack (object): The object to be searched for the compared value
+    """
+    def __init__(self, haystack):
```

Comment:
The argument to `Needle()` should be _the needle_, not the haystack.
The usage is intended to be something like this.

Actual call:

```python
my_method('Lorem ipsum dolor set amet blah blah blah blah...')
```

Proving it in the mock:

```python
mock_method.assert_called_once_with(duck.Needle('Lorem ipsum'))
```